staging/src/k8s.io/component-base/config/testing/roundtrip.go

```diff
@@ -90,3 +92,89 @@ func GetRoundtripTestCases(scheme *runtime.Scheme, disallowMarshalGroupVersions
 	}
 	return cases
 }
+
+func runRoundTripTests(t *testing.T, tests []TestCase, codecs serializer.CodecFactory) {
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			object := decodeYAML(t, tc.in, tc.codec)
+
+			roundTrip(t, tc.codec, object)
+
+			data, err := runtime.Encode(tc.codec, object)
+			if err != nil {
+				t.Fatalf("failed to encode object: %v", err)
+				return
+			}
+
+			matchOutputFile(t, tc.out, data)
+		})
+	}
+}
+
+func roundTrip(t *testing.T, codec runtime.Codec, object runtime.Object) {
+	original := object
+
+	// deep copy the original object
+	object = original.DeepCopyObject()
+	if !apiequality.Semantic.DeepEqual(original, object) {
+		t.Errorf("deepcopy altered the original object, diff (- want, + got):\n%v", cmp.Diff(original, object))
+		return
+	}
+
+	// encode (serialize) the deep copy using the provided codec
+	data, err := runtime.Encode(codec, object)
+	if err != nil {
+		t.Errorf("failed to encode object: %v", err)
+		return
+	}
+
+	// ensure that the deep copy is equal to the original; neither the deep
```

Comment:
Update the comment to reflect that this is checking that encode didn't change the object, not the deep copy.